### PR TITLE
Extend umb-checkbox to allow icons

### PR DESCRIPTION
**Pull request by BatJan**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description
I noticed that the rich text editor datatype was using the umb-checkbox directive but that there was a `<label>` wrapped around it, which should not be neccessary. But it was in order to ensure that clicking the associated config icon and text would tick the box.

Now that's not needed anymore since I have enabled the directive to accept an `iconClass` parameter, and when it's passed a value it will be rendered. if it's left empty it will not. Then things will look and act like we're used to.

Doing this fixes the markup pattern so we don't have `<label>` nested inside a `<label>` - Visually things look the same as before.

This can be tested in the "rich text editor" datatype 👍

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/views/components/forms/umb-checkbox.html [hunk 1]
```diff
@@ -16,5 +16,12 @@
         </span>
     </span>
 
+    <i class="{{vm.iconClass}}" style="margin-left: 26px;" aria-hidden="true"></i>
+
     <span class="umb-form-check__text">{{vm.text}}</span>
 </label>
+
+
+<!-- TODO
```
Comment (poornimanayar) [inline src/Umbraco.Web.UI.Client/src/views/components/forms/umb-checkbox.html]: Is this note needed here as you seem to have handled this in a further commit?

Comment (BatJan) [reply src/Umbraco.Web.UI.Client/src/views/components/forms/umb-checkbox.html]: Oops! Removed now :-)

**Comment by BatJan** [comment]
@poornimanayar - Thank you for testing 😃 Removed the comment 👍

**Comment by poornimanayar** [comment]
No worries my friend :-) Happy to help!

**Comment by poornimanayar** [comment]
Thanks @BatJan ! Merged :-)

**Pull request merged**